src/v/coproc/script_context_router.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/partition.h"
+#include "model/fundamental.h"
+
+#include <seastar/core/shared_ptr.hh>
+
+#include <absl/container/node_hash_map.h>
+
+namespace coproc {
+
+/// Read state for a single input topic, unique and unshared
+struct read_context {
+    static constexpr int8_t current_version{1};
+    /// Manual version handling of structure
+    int8_t version{current_version};
+    /// Where to begin reading from, determined by user provided option \ref
+    /// topic_ingestion_policy, earliest, latest, or stored
+    model::offset absolute_start{};
+    /// Highest offset of current in-progress read
+    model::offset last_read{};
+    /// Highest offset that all materialized topics have observed
+    model::offset last_acked{};
+    /// Pointer to an input topic
+    ss::lw_shared_ptr<cluster::partition> input;
+};
+
+/// Write state for topics that have been created stemming from an associated
+/// source, the input begint the pointer in \ref read_context
+struct write_context {
+    static constexpr int8_t current_version{1};
+    /// Manual version handling of structure
+    int8_t version{current_version};
```

Comment:
why is `version` a non-static data member?

Reply:
Ok modified this after `data_policy.h` it is now static